### PR TITLE
tool: (irdl-to-pyrdl) Generate better class name

### PR DESCRIPTION
**Pull request by alexarice**
Makes operation class names generated by `irdl_to_pyrdl` be in CamelCase and end with Op.

Unfortunately it seems this is not the tool that deals with `cmath.irdl` :facepalm: 

Related to #3527

### CODE REVIEW

File: xdsl/dialects/irdl/irdl_to_pyrdl.py [hunk 1]
```diff
@@ -33,11 +33,15 @@ class {op.sym_name.data}(ParametrizedAttribute{type_addition}):
     return res
 
 
+def get_op_name(op: OperationOp) -> str:
```
Comment (superlopuh) [inline xdsl/dialects/irdl/irdl_to_pyrdl.py:36]:
```suggestion
def class_name_from_op_name(op: OperationOp) -> str:
```

Comment (superlopuh) [reply xdsl/dialects/irdl/irdl_to_pyrdl.py:36]: I would recommend adding a separate unit test for this, and to handle the case `dialect.part1.part2_bla`

Comment (alexarice) [reply xdsl/dialects/irdl/irdl_to_pyrdl.py]: I want to move it somewhere more generic to reuse in the next PR, is `dialects/irdl/utils.py` a decent place?

Comment (alexarice) [reply xdsl/dialects/irdl/irdl_to_pyrdl.py]: Can a symbol name have a `.` in it?

Comment (alexarice) [reply xdsl/dialects/irdl/irdl_to_pyrdl.py]: Seems so

**Review by superlopuh — APPROVED** [review]

**Review by math-fehr — APPROVED** [review]
Cool!

**Comment by alexarice** [comment]
Added suggestion from #3529 (sorry this got so confusing)

**Review by superlopuh — APPROVED** [review]

**Pull request merged**